app/controllers/dashboard/form/publish_controller.rb

```diff
@@ -38,7 +38,7 @@ def update
         # actually attempt it. However, we want to be nice and not yell at them for something they _haven't_ seen yet
         # like rights.
         def prevalidate
-          @resource.publish
+          @resource.publish unless @resource.published?
```

Comment:
I think I would like to refactor this method to temporarily publish the @resource, and then roll it back to whatever its initial state was. That would, I think, prevent some of the weirdness in the views.